### ISSUE TITLE
Error reporter

### ISSUE THREAD

**Issue by ice1000** [rikvdkleij/intellij-haskell#351]
When IDE throws an Exception, it has the ability to blame on a plugin.
It also supports reporting the Exception with the stacktrace to somewhere when the Exception occurs.

Any interests in adding this feature (and the error will be reported to GitHub, with a format like ice1000/julia-intellij#270)?
I know how to impl this, just asking a question here.

**Comment by rikvdkleij** [comment]
Cool! Very helpful.

**Comment by ice1000** [comment]
:+1:

**Comment by develop7** [comment]
OHMYGOD YES PLEASE

consider wrapping stacktrace to something like `<details><summary>Something meaningful</summary>...stacktrace...</details>` BTW. Works good for https://github.com/KronicDeth/intellij-elixir/issues/902

**Comment by ice1000** [comment]
> consider wrapping stacktrace to something like <details><summary>Something meaningful</summary>...stacktrace...</details> BTW. Works good for KronicDeth/intellij-elixir#902

OHMYGOD THAT"S AWESOME

**Comment by ice1000** [comment]
Oh my god I just realized I'll have to rewrite my old error reporter in Kotlin to Scala

**Comment by ice1000** [comment]
Can I use Kotlin?
It's too hard to deal with Java collections in Scala codes.

**Comment by ice1000** [comment]
[intellij-haskell.zip](https://github.com/rikvdkleij/intellij-haskell/files/2689411/intellij-haskell.zip)

**Comment by rikvdkleij** [comment]
> Can I use Kotlin?

Well, I prefer Scala but I have no problem with Kotlin itself. I do not know how to integrate with sbt build and IntelliJ scala project.

> It's too hard to deal with Java collections in Scala codes.

It's not difficult, you just have to import:
`import scala.collection.JavaConverters._`

**Comment by ice1000** [comment]
> Well, I prefer Scala but I have no problem with Kotlin itself. I do not know how to integrate with sbt build and IntelliJ scala project.

No need to worry about this. IntelliJ will help with compiling Kotlin codes during packing the .zip.

**Comment by rikvdkleij** [comment]
And sbt compile/build?

**Comment by rikvdkleij** [comment]
I would appreciate if you try solution with:
`import scala.collection.JavaConverters._`

Maybe it's already  implemented in IntelliJ Scala plugin.

**Comment by ice1000** [comment]
> And sbt compile/build?

Not sure, have 0 knowledge about this

**Comment by ice1000** [comment]
Yes, will try later. But you may try the .zip I've just uploaded, it's nice.

**Comment by ice1000** [comment]
Did you try the language injection feature? It's great when you're working with regex or jsons

**Comment by rikvdkleij** [comment]
> Did you try the language injection feature? 

Not yet. I will try later.

**Comment by ice1000** [comment]
> Maybe it's already implemented in IntelliJ Scala plugin.

They use YouTrack, we use GitHub. It's different.
